batchflow/models/torch/encoder_decoder.py

```diff
@@ -87,29 +83,23 @@ def forward(self, x):
 
 
 
-class DecoderModule(nn.Module):
+class DecoderModule(nn.ModuleDict):
     """ Decoder: increasing spatial dimensions. """
     def __init__(self, inputs=None, **kwargs):
         super().__init__()
         self._make_modules(inputs, **kwargs)
 
     def forward(self, x):
-        b_counter, u_counter, c_counter = 0, 0, 0
         inputs = x if isinstance(x, list) else [x]
         x = inputs[-1]
-
-        for i in range(self.num_stages):
-            for letter in self.decoder_layout:
-                if letter in ['b']:
-                    x = self.decoder_b[b_counter](x)
-                    b_counter += 1
-                elif letter in ['u']:
-                    x = self.decoder_u[u_counter](x)
-                    u_counter += 1
-                elif letter in ['c']:
-                    if self.skip and (i < len(inputs) - 2):
-                        x = self.decoder_c[c_counter]([x, inputs[-i - 3]])
-                        c_counter += 1
+        i = 0
+
+        for letter, layer in zip(self.layout, self.values()):
+            if letter in ['b', 'u']:
+                x = layer(x)
+            elif letter in ['c'] and self.skip and (i < len(inputs) - 2):
+                x = layer([x, inputs[-i - 3]])
+                i += 1
         return x
```

Comment:
what if letter not in `bcu`?

Reply:
Then the error will be raised during the making of the module, before `forward`